tools/pkglint/lib/rules.ts

```diff
@@ -173,34 +173,69 @@ export class StabilitySetting extends ValidationRule {
       return;
     }
 
-    let stability = pkg.json.stability;
-    switch (stability) {
-      case 'experimental':
-      case 'stable':
-      case 'deprecated':
-        if (pkg.json.deprecated && stability !== 'deprecated') {
-          pkg.report({
-            ruleName: this.name,
-            message: `Package is deprecated, but is marked with stability "${stability}"`,
-            fix: () => pkg.json.stability = 'deprecated',
-          });
-          stability = 'deprecated';
-        }
-        break;
-      default:
-        const defaultStability = pkg.json.deprecated ? 'deprecated' : 'experimental';
-        pkg.report({
-          ruleName: this.name,
-          message: `Invalid stability configuration in package.json: ${JSON.stringify(stability)}`,
-          fix: () => pkg.json.stability = defaultStability,
-        });
-        stability = defaultStability;
+    // 'stability' contains the default API-level stability, either 'experimental', 'stable',
+    // 'deprecated' or 'external'. This is the value that will be used for every class
+    // unless it is annotated otherwise. It must be one of these values, which will be
+    // enforced by 'jsii'.
+    //
+    // On top of this, we have conceptual library-level "maturity", which is one of
+    // the following stages: 'cfn-only', 'experimental', 'developer-preview', 'stable',
+    // 'deprecated'. These render into stability banners in the README.
+    //
+    // The maturities mostly map onto default stabilities, and so maturity will
+    // be derived from stability if possible. The mapping is:
```

Comment:
Feels like stability should derive from maturity, no?

Experimental, Developer preview => experimental
Stable => Stable

CFN classes should always be annotated with @external, no?

Reply:
Yeah, I guess that makes more sense. I was doing the minimal change from what we currently have. 

> CFN classes should always be annotated with @external, no?

And so they are. Still, "external" is a valid stability and hence can be put into the default "stability" field.